### PR TITLE
IR generation for sha256.

### PR DESCRIPTION
**Pull request by chriseth**
Part of #8343.

### CODE REVIEW

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -1474,9 +1503,15 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 		// but all parameters of ecrecover are value types anyway.
 		encodeInPlace = false;
 	bool encodeForLibraryCall = funKind == FunctionType::Kind::DelegateCall;
-	solUnimplementedAssert(!encodeInPlace, "");
-	solUnimplementedAssert(funType.padArguments(), "");
```
Comment (axic) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:1552]: Isn't it worth keeping this assert for the `!encodeInPlace` case?

Comment (chriseth) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:1552]: You mean like that: https://github.com/ethereum/solidity/pull/8763/files#diff-9e73c6daf061b8b0e6dff4531c75e83eR1539

**Comment by chriseth** [comment]
Fixed.

**Comment by axic** [comment]
Sorry, can you change the commit message to reflect ripemd160+ecrecover too ?

**Review by axic — APPROVED** [review]
Apart from the commit message looks good.

**Comment by chriseth** [comment]
ecrecover and ripemd are still disabled in `appendExternalFunctionCall` because they still need some more code.

**Comment by axic** [comment]
You mean they are disabled through some of the encoding assertions?

**Comment by chriseth** [comment]
No, through function kind assertion.

**Comment by axic** [comment]
I'm blind but can't see it. Which line is it?

**Comment by chriseth** [comment]
At the very end - not the best place, but it was already there :)

**Comment by axic** [comment]
Ah found it. Actually there are two assertions for ecrecover. This is rather confusing, glad we clarified :wink:

**Comment by axic** [comment]
Good to go then.

**Pull request merged**